src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -673,20 +673,35 @@ static bool parseMemLimit(const char* line, intx& value, int& bytes_read, char*
   return true;
 }
 
-static bool parseEnumValueAsUintx(enum CompileCommand option, const char* line, uintx& value, int& bytes_read, char* errorbuf, const int buf_size) {
+// Parse an uintx-based option value. Also takes care of parsing enum values for options that are enums.
+// Returns true if ok, false if the value could not be parsed.
+static bool parseUintxValue(enum CompileCommand option, const char* line, uintx& value, int& bytes_read) {
```

Comment:
It is honestly weird to see `parse***Uintx***Value` dealing with enums, and be specialized for `MemStat`. Can you reflow this to match how `MemLimit` does it? https://github.com/openjdk/jdk/blob/7bb1999c51cdfeb020047e1094229fda1ec5a99d/src/hotspot/share/compiler/compilerOracle.cpp#L702

Reply:
Okay, rewritten in the style of https://github.com/openjdk/jdk/pull/16631. Retested too. Let's get this out of the door, please.